File: src/my_project/medium_problems/from51to100/watering_plants.py
from typing import List 

class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        
        return_count: int  = 0
        temp_capacity: int  = capacity 
        
        len_plants: int = len(plants)
        
        for i in range(len_plants):
            if temp_capacity < plants[i]:
                return_count += i*2
                temp_capacity = capacity - plants[i]
            else: 
                temp_capacity -= plants[i]
                
        
        return len_plants + return_count

File: src/my_project_typescript/medium_problems_typescript/wateringPlants.ts
function wateringPlants(plants: number[], capacity: number): number {

    let returnCount: number = 0
    let tempCapacity: number = capacity

    let lenPlants: number = plants.length

    for (let i = 0; i < lenPlants; i++){
        if (tempCapacity < plants[i]){
            returnCount += i*2
            tempCapacity = capacity - plants[i]
        }
        else {
            tempCapacity -= plants[i]
        }
    }

    return lenPlants + returnCount

};


function wateringPlantsTwo(plants: number[], capacity: number): number {

    let returnCount: number = 0
    let tempCapacity: number = capacity

    let lenPlants: number = plants.length

    plants.forEach( (element, index) => {
        if (tempCapacity < element){
            returnCount += index*2
            tempCapacity = capacity - element
        }
        else {
            tempCapacity -= element
        }
    })

    return lenPlants + returnCount

};